<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Nur Aziz" />


<title>Gene Expression Note 1</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.23/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.11.3/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.11.3/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.11.3/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="css/one-column-cv.css" type="text/css" />
<link rel="stylesheet" href="css/website-pagetable.css" type="text/css" />
<link rel="stylesheet" href="css/style.css" type="text/css" />
<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">About Me</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="wet-lab-notebook.html">Wet Lab Notebook</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Dry Lab Notebook
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="PRMT1-RNAseq-recovery.html">RNAseq analysis</a>
    </li>
    <li>
      <a href="METTL21B.html">Plotting METTL21B</a>
    </li>
    <li>
      <a href="Gene-Expression-Note-1.html">Gene Expression Note 1</a>
    </li>
    <li>
      <a href="Gene-Expression-Note-2.html">Gene Expression Note 2</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Gene Expression Note 1</h1>
<h4 class="author">Nur Aziz</h4>

</div>


<div id="profiling-gene-expression-from-geo-tcga-gtex"
class="section level1" number="1">
<h1><span class="header-section-number">1</span> Profiling Gene
Expression from GEO, TCGA, GTEx</h1>
<p>Before begin, it’s recommended to read these publications:</p>
<ul>
<li><a href="https://doi.org/10.1016/j.xpro.2021.100478">Analysis
workflow of publicly available RNA-sequencing datasets</a></li>
<li><a href="https://doi.org/10.12688/f1000research.8923.2">TCGA
Workflow: Analyze cancer genomics and epigenomics data using
Bioconductor packages</a></li>
<li><a href="https://doi.org/10.1093/nar/gkv1507">TCGAbiolinks: an
R/Bioconductor package for integrative analysis of TCGA data</a></li>
<li><a href="https://doi.org/10.1371/journal.pcbi.1006701">New
functionalities in the TCGAbiolinks package for the study and
integration of cancer data from GDC and GTEx</a></li>
<li><a href="https://doi.org/10.1038/sdata.2018.61">Unifying cancer and
normal RNA sequencing data from different sources</a></li>
</ul>
<blockquote>
<p>We will make similar results like these : <img
src="docs/Gene_Exp_1/example%20gse%201.png" alt="Example 1" /> <img
src="docs/Gene_Exp_1/example%20gse%202.png" alt="Example 2" /></p>
</blockquote>
<div id="geo" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> GEO</h2>
<div id="normal-vs-tumor" class="section level3" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> Normal vs
Tumor</h3>
<p>Install and load libraries:</p>
<pre class="r"><code>Sys.setenv(LANG=&#39;en&#39;) #change warning to English 
Sys.setenv(&quot;VROOM_CONNECTION_SIZE&quot; = 131072 * 10) #prevent connection lost when roaming big chunk
library(GEOquery)
library(limma)
library(tidyverse)
library(dplyr)
library(&quot;ggsci&quot;)
library(rstatix)

create_dt&lt;-function(x){
  DT::datatable(x,
                extensions = &#39;Buttons&#39;,
                options = list(dom=&#39;Blfrtip&#39;,
                               buttons=c(&#39;copy&#39;, &#39;csv&#39;, &#39;excel&#39;,&#39;pdf&#39;, &#39;print&#39;),
                               lengthMenu=list(c(10,25,50,-1),
                                               c(10,25,50,&#39;All&#39;))))
}</code></pre>
<p>To make graph look nice and consistents, I make my own function : *
Make color palette for your graph</p>
<pre class="r"><code>library(scales)
pallet&lt;-c(&quot;#386cb0&quot;,&quot;#fdb462&quot;,&quot;#7fc97f&quot;,&quot;#a6cee3&quot;,&quot;#fb9a99&quot;,&quot;#984ea3&quot;,&quot;#ffff33&quot;)
show_col(pallet)</code></pre>
<p><img src="Gene-Expression-Note-1_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<pre class="r"><code>####Aziz&#39; function&#39;s theme and color palettes

scale_fill_aziz &lt;- function(...){
  library(scales)
  discrete_scale(&quot;fill&quot;,&quot;aziz&quot;,manual_pal(values = c(&quot;#386cb0&quot;,&quot;#fdb462&quot;,&quot;#7fc97f&quot;,&quot;#a6cee3&quot;,&quot;#fb9a99&quot;,&quot;#984ea3&quot;,&quot;#ffff33&quot;)), ...)
  
}
scale_colour_aziz &lt;- function(...){
  library(scales)
  discrete_scale(&quot;colour&quot;,&quot;aziz&quot;,manual_pal(values = c(&quot;#386cb0&quot;,&quot;#fdb462&quot;,&quot;#7fc97f&quot;,&quot;#ef3b2c&quot;,&quot;#662506&quot;,&quot;#a6cee3&quot;,&quot;#fb9a99&quot;,&quot;#984ea3&quot;,&quot;#ffff33&quot;)), ...)
} 
theme_Publication &lt;- function(base_size=14, base_family=&quot;helvetica&quot;) {
  library(grid)
  library(ggthemes)
  (theme_foundation(base_size=base_size, base_family=base_family)
    + theme(plot.title = element_text(face = &quot;bold&quot;,
                                      size = rel(1.2), hjust = 0.5),
            text = element_text(),
            panel.background = element_rect(colour = NA),
            plot.background = element_rect(colour = NA),
            panel.border = element_rect(colour = NA),
            axis.title = element_text(face = &quot;bold&quot;,size = rel(1)),
            axis.title.y = element_text(angle=90,vjust =2),
            axis.title.x = element_text(vjust = -0.2),
            axis.text = element_text(), 
            axis.line = element_line(colour=&quot;black&quot;),
            axis.ticks = element_line(),
            panel.grid.major = element_line(colour=&quot;#f0f0f0&quot;),
            panel.grid.minor = element_blank(),
            legend.key = element_rect(colour = NA),
            legend.position = &quot;bottom&quot;,
            legend.direction = &quot;horizontal&quot;,
            legend.key.size= unit(0.2, &quot;cm&quot;),
            legend.margin = unit(0, &quot;cm&quot;),
            legend.title = element_text(face=&quot;italic&quot;),
            plot.margin=unit(c(10,5,5,5),&quot;mm&quot;),
            strip.background=element_rect(colour=&quot;#f0f0f0&quot;,fill=&quot;#f0f0f0&quot;),
            strip.text = element_text(face=&quot;bold&quot;)
    ))
  
}</code></pre>
<p><br></p>
<p>For example using ACRG dataset</p>
<pre class="r"><code>#### GSE62254 (ACRG)
gset &lt;- getGEO(&quot;GSE66229&quot;, GSEMatrix =TRUE, AnnotGPL=TRUE)
if (length(gset) &gt; 1) idx &lt;- grep(&quot;GPL570&quot;, attr(gset, &quot;names&quot;)) else idx &lt;- 1
gset &lt;- gset[[idx]]

# make proper column names to match toptable 
fvarLabels(gset) &lt;- make.names(fvarLabels(gset))

# group membership for all samples
gsms &lt;- paste0(&quot;11100000000000000111110000000000000000100111111111&quot;,
               &quot;11111111111111111111111111111111111111111111111111&quot;,
               &quot;11111111111111111111111111111111111111111111111111&quot;,
               &quot;11111111111111111111111010111101111000000111111111&quot;,
               &quot;11111111111111111111111111111111111111111111111111&quot;,
               &quot;11111111111111111111111111111111111111111111111111&quot;,
               &quot;11111111111111111111111111111111111100000111110000&quot;,
               &quot;00000000000000000000000000000000000000000000000000&quot;)
sml &lt;- strsplit(gsms, split=&quot;&quot;)[[1]]

# log2 transformation
ex &lt;- exprs(gset)
qx &lt;- as.numeric(quantile(ex, c(0., 0.25, 0.5, 0.75, 0.99, 1.0), na.rm=T))
LogC &lt;- (qx[5] &gt; 100) ||
  (qx[6]-qx[1] &gt; 50 &amp;&amp; qx[2] &gt; 0)
if (LogC) { ex[which(ex &lt;= 0)] &lt;- NaN
exprs(gset) &lt;- log2(ex) }

# assign samples to groups and set up design matrix
gs &lt;- factor(sml)
groups &lt;- make.names(c(&quot;N&quot;,&quot;T&quot;))
levels(gs) &lt;- groups
gset$group &lt;- gs
design &lt;- model.matrix(~group + 0, gset)
colnames(design) &lt;- levels(gs)

fit &lt;- lmFit(gset, design)  # fit linear model

# set up contrasts of interest and recalculate model coefficients
cts &lt;- paste(groups[1], groups[2], sep=&quot;-&quot;)
cont.matrix &lt;- makeContrasts(contrasts=cts, levels=design)
fit2 &lt;- contrasts.fit(fit, cont.matrix)

# compute statistics and table of top significant genes
fit2 &lt;- eBayes(fit2, 0.01)
tT &lt;- topTable(fit2, adjust=&quot;fdr&quot;, sort.by=&quot;B&quot;, number=250)

tT &lt;- subset(tT, select=c(&quot;ID&quot;,&quot;adj.P.Val&quot;,&quot;P.Value&quot;,&quot;t&quot;,&quot;B&quot;,&quot;logFC&quot;,&quot;Gene.symbol&quot;,&quot;Gene.title&quot;))

# Visualize and quality control test results.
# Build histogram of P-values for all genes. Normal test
# assumption is that most genes are not differentially expressed.
tT2 &lt;- topTable(fit2, adjust=&quot;fdr&quot;, sort.by=&quot;B&quot;, number=Inf)
hist(tT2$adj.P.Val, col = &quot;grey&quot;, border = &quot;white&quot;, xlab = &quot;P-adj&quot;,
     ylab = &quot;Number of genes&quot;, main = &quot;P-adj value distribution&quot;)</code></pre>
<p><img src="Gene-Expression-Note-1_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre class="r"><code>#prepare for only genes passing the threshold
newtT2&lt;-tT2 %&gt;% 
  filter(adj.P.Val&lt;0.05)

newtT3 &lt;- subset(newtT2,select=c(&quot;ID&quot;,&quot;adj.P.Val&quot;,&quot;t&quot;,&quot;B&quot;,&quot;logFC&quot;,&quot;Gene.symbol&quot;,&quot;Gene.title&quot;)) #useful for DEGs analysis, but we don&#39;t need right now #skip this

#retrieve the expression of your gene in all patient
threegenes&lt;-ex[row.names(ex) %in% c(&quot;219617_at&quot;, #camkmt
                                    &quot;222013_x_at&quot;, #FAM86A
                                    &quot;206445_s_at&quot;),] #PRMT1
head(threegenes)[,1:5]</code></pre>
<pre><code>##             GSM1523727 GSM1523728 GSM1523729 GSM1523730 GSM1523731
## 206445_s_at   2.957375   3.128566   3.052887   2.916911   2.789613
## 219617_at     1.999631   1.689934   1.806846   1.659223   1.710085
## 222013_x_at   1.758549   1.830104   1.791574   1.750806   1.744396</code></pre>
<pre class="r"><code>head(t(threegenes)) #transpose the table</code></pre>
<pre><code>##            206445_s_at 219617_at 222013_x_at
## GSM1523727    2.957375  1.999631    1.758549
## GSM1523728    3.128566  1.689934    1.830104
## GSM1523729    3.052887  1.806846    1.791574
## GSM1523730    2.916911  1.659223    1.750806
## GSM1523731    2.789613  1.710085    1.744396
## GSM1523732    3.088736  1.715045    1.766945</code></pre>
<pre class="r"><code>threegenes&lt;-t(threegenes) 
GOI&lt;-as.data.frame(threegenes) #GOI = gene of interest
GOI&lt;-GOI %&gt;% 
  mutate(Patient.ID=row.names(threegenes))
row.names(GOI)&lt;-NULL
GOI&lt;-GOI %&gt;% 
  select(4,1,2,3)
head(GOI)</code></pre>
<pre><code>##   Patient.ID 206445_s_at 219617_at 222013_x_at
## 1 GSM1523727    2.957375  1.999631    1.758549
## 2 GSM1523728    3.128566  1.689934    1.830104
## 3 GSM1523729    3.052887  1.806846    1.791574
## 4 GSM1523730    2.916911  1.659223    1.750806
## 5 GSM1523731    2.789613  1.710085    1.744396
## 6 GSM1523732    3.088736  1.715045    1.766945</code></pre>
<pre class="r"><code>colnames(GOI)[2]&lt;-&quot;PRMT1&quot;
colnames(GOI)[3]&lt;-&quot;CAMKMT&quot;
colnames(GOI)[4]&lt;-&quot;FAM86A&quot;

# download phenotypes
my.gse &lt;- &quot;GSE66229&quot;

if(!file.exists(&quot;geo_downloads&quot;)) dir.create(&quot;geo_downloads&quot;)
if(!file.exists(&quot;results&quot;))  dir.create(&quot;results&quot;, recursive=TRUE)
my.geo.gse &lt;- getGEO(GEO=my.gse, filename=NULL, destdir=&quot;./geo_downloads&quot;, GSElimits=NULL, GSEMatrix=TRUE, AnnotGPL=FALSE, getGPL=FALSE)
my.geo.gse &lt;- my.geo.gse[[1]]
my.pdata &lt;- as.data.frame(pData(my.geo.gse), stringsAsFactors=F)
head(my.pdata)</code></pre>
<pre><code>##            title geo_accession                status submission_date
## GSM1523727  T107    GSM1523727 Public on Apr 17 2015     Oct 10 2014
## GSM1523728  T108    GSM1523728 Public on Apr 17 2015     Oct 10 2014
## GSM1523729  T109    GSM1523729 Public on Apr 17 2015     Oct 10 2014
## GSM1523730  N132    GSM1523730 Public on Feb 15 2019     Oct 10 2014
## GSM1523731  N133    GSM1523731 Public on Feb 15 2019     Oct 10 2014
## GSM1523732  N134    GSM1523732 Public on Feb 15 2019     Oct 10 2014
##            last_update_date type channel_count         source_name_ch1
## GSM1523727      Apr 17 2015  RNA             1  Tumor from patient 107
## GSM1523728      Apr 17 2015  RNA             1  Tumor from patient 108
## GSM1523729      Apr 17 2015  RNA             1  Tumor from patient 109
## GSM1523730      Feb 15 2019  RNA             1 Normal from patient 132
## GSM1523731      Feb 15 2019  RNA             1 Normal from patient 133
## GSM1523732      Feb 15 2019  RNA             1 Normal from patient 134
##            organism_ch1           characteristics_ch1 characteristics_ch1.1
## GSM1523727 Homo sapiens         tissue: Gastric tumor          patient: 107
## GSM1523728 Homo sapiens         tissue: Gastric tumor          patient: 108
## GSM1523729 Homo sapiens         tissue: Gastric tumor          patient: 109
## GSM1523730 Homo sapiens tissue: normal gastric tissue          patient: 132
## GSM1523731 Homo sapiens tissue: normal gastric tissue          patient: 133
## GSM1523732 Homo sapiens tissue: normal gastric tissue          patient: 134
##            molecule_ch1
## GSM1523727    total RNA
## GSM1523728    total RNA
## GSM1523729    total RNA
## GSM1523730    total RNA
## GSM1523731    total RNA
## GSM1523732    total RNA
##                                                                                                                                                                                                  extract_protocol_ch1
## GSM1523727 Samples are processed in parallel in 96-well plates to minimize potential variation. Reaction purification is achieved using magnetic binding beads for cDNA and Qiagen RNeasy kits for cRNA purification.
## GSM1523728 Samples are processed in parallel in 96-well plates to minimize potential variation. Reaction purification is achieved using magnetic binding beads for cDNA and Qiagen RNeasy kits for cRNA purification.
## GSM1523729 Samples are processed in parallel in 96-well plates to minimize potential variation. Reaction purification is achieved using magnetic binding beads for cDNA and Qiagen RNeasy kits for cRNA purification.
## GSM1523730 Samples are processed in parallel in 96-well plates to minimize potential variation. Reaction purification is achieved using magnetic binding beads for cDNA and Qiagen RNeasy kits for cRNA purification.
## GSM1523731 Samples are processed in parallel in 96-well plates to minimize potential variation. Reaction purification is achieved using magnetic binding beads for cDNA and Qiagen RNeasy kits for cRNA purification.
## GSM1523732 Samples are processed in parallel in 96-well plates to minimize potential variation. Reaction purification is achieved using magnetic binding beads for cDNA and Qiagen RNeasy kits for cRNA purification.
##            label_ch1           label_protocol_ch1 taxid_ch1
## GSM1523727    biotin standard Affymetrix protocol      9606
## GSM1523728    biotin standard Affymetrix protocol      9606
## GSM1523729    biotin standard Affymetrix protocol      9606
## GSM1523730    biotin standard Affymetrix protocol      9606
## GSM1523731    biotin standard Affymetrix protocol      9606
## GSM1523732    biotin standard Affymetrix protocol      9606
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                hyb_protocol
## GSM1523727 GeneChip microarrays are loaded with the fragmented target sample/hybridization buffer mix using standard manual techniques. Arrays are hybridized for 18 hours at 45oC with vigorous mixing. Unbound sample is removed and staining is accomplished through the binding of streptavidin conjugated phycoerythrin to the hybridized target. Excess label is removed. Washing and staining steps are performed by the Affymetrix FS450 fluidics station using standard protocols.
## GSM1523728 GeneChip microarrays are loaded with the fragmented target sample/hybridization buffer mix using standard manual techniques. Arrays are hybridized for 18 hours at 45oC with vigorous mixing. Unbound sample is removed and staining is accomplished through the binding of streptavidin conjugated phycoerythrin to the hybridized target. Excess label is removed. Washing and staining steps are performed by the Affymetrix FS450 fluidics station using standard protocols.
## GSM1523729 GeneChip microarrays are loaded with the fragmented target sample/hybridization buffer mix using standard manual techniques. Arrays are hybridized for 18 hours at 45oC with vigorous mixing. Unbound sample is removed and staining is accomplished through the binding of streptavidin conjugated phycoerythrin to the hybridized target. Excess label is removed. Washing and staining steps are performed by the Affymetrix FS450 fluidics station using standard protocols.
## GSM1523730 GeneChip microarrays are loaded with the fragmented target sample/hybridization buffer mix using standard manual techniques. Arrays are hybridized for 18 hours at 45oC with vigorous mixing. Unbound sample is removed and staining is accomplished through the binding of streptavidin conjugated phycoerythrin to the hybridized target. Excess label is removed. Washing and staining steps are performed by the Affymetrix FS450 fluidics station using standard protocols.
## GSM1523731 GeneChip microarrays are loaded with the fragmented target sample/hybridization buffer mix using standard manual techniques. Arrays are hybridized for 18 hours at 45oC with vigorous mixing. Unbound sample is removed and staining is accomplished through the binding of streptavidin conjugated phycoerythrin to the hybridized target. Excess label is removed. Washing and staining steps are performed by the Affymetrix FS450 fluidics station using standard protocols.
## GSM1523732 GeneChip microarrays are loaded with the fragmented target sample/hybridization buffer mix using standard manual techniques. Arrays are hybridized for 18 hours at 45oC with vigorous mixing. Unbound sample is removed and staining is accomplished through the binding of streptavidin conjugated phycoerythrin to the hybridized target. Excess label is removed. Washing and staining steps are performed by the Affymetrix FS450 fluidics station using standard protocols.
##                                                                                                                                                                         scan_protocol
## GSM1523727 Arrays are scanned using a GeneChip Scanner 3000 7G with a 48 array autoloader. The scanner maintains the optimal temperature for the arrays prior to and during scanning.
## GSM1523728 Arrays are scanned using a GeneChip Scanner 3000 7G with a 48 array autoloader. The scanner maintains the optimal temperature for the arrays prior to and during scanning.
## GSM1523729 Arrays are scanned using a GeneChip Scanner 3000 7G with a 48 array autoloader. The scanner maintains the optimal temperature for the arrays prior to and during scanning.
## GSM1523730 Arrays are scanned using a GeneChip Scanner 3000 7G with a 48 array autoloader. The scanner maintains the optimal temperature for the arrays prior to and during scanning.
## GSM1523731 Arrays are scanned using a GeneChip Scanner 3000 7G with a 48 array autoloader. The scanner maintains the optimal temperature for the arrays prior to and during scanning.
## GSM1523732 Arrays are scanned using a GeneChip Scanner 3000 7G with a 48 array autoloader. The scanner maintains the optimal temperature for the arrays prior to and during scanning.
##                                                                                                                                    description
## GSM1523727 ACRG Gastric cohort: microarray profiles from 300 gastric tumors and 100 patient matched normal tissue from gastric cancer patients
## GSM1523728 ACRG Gastric cohort: microarray profiles from 300 gastric tumors and 100 patient matched normal tissue from gastric cancer patients
## GSM1523729 ACRG Gastric cohort: microarray profiles from 300 gastric tumors and 100 patient matched normal tissue from gastric cancer patients
## GSM1523730 ACRG Gastric cohort: microarray profiles from 300 gastric tumors and 100 patient matched normal tissue from gastric cancer patients
## GSM1523731 ACRG Gastric cohort: microarray profiles from 300 gastric tumors and 100 patient matched normal tissue from gastric cancer patients
## GSM1523732 ACRG Gastric cohort: microarray profiles from 300 gastric tumors and 100 patient matched normal tissue from gastric cancer patients
##                           description.1
## GSM1523727 Nugen amplification protocol
## GSM1523728 Nugen amplification protocol
## GSM1523729 Nugen amplification protocol
## GSM1523730 Nugen amplification protocol
## GSM1523731 Nugen amplification protocol
## GSM1523732 Nugen amplification protocol
##                                                                                                          data_processing
## GSM1523727 The data were analyzed by RMA with Affymetrix Power Tools package using Affymetrix default analysis settings.
## GSM1523728 The data were analyzed by RMA with Affymetrix Power Tools package using Affymetrix default analysis settings.
## GSM1523729 The data were analyzed by RMA with Affymetrix Power Tools package using Affymetrix default analysis settings.
## GSM1523730 The data were analyzed by RMA with Affymetrix Power Tools package using Affymetrix default analysis settings.
## GSM1523731 The data were analyzed by RMA with Affymetrix Power Tools package using Affymetrix default analysis settings.
## GSM1523732 The data were analyzed by RMA with Affymetrix Power Tools package using Affymetrix default analysis settings.
##            platform_id      contact_name                contact_department
## GSM1523727      GPL570 Michael,,Nebozhyn GpGx/Computational System biology
## GSM1523728      GPL570 Michael,,Nebozhyn GpGx/Computational System biology
## GSM1523729      GPL570 Michael,,Nebozhyn GpGx/Computational System biology
## GSM1523730      GPL570 Michael,,Nebozhyn GpGx/Computational System biology
## GSM1523731      GPL570 Michael,,Nebozhyn GpGx/Computational System biology
## GSM1523732      GPL570 Michael,,Nebozhyn GpGx/Computational System biology
##            contact_institute
## GSM1523727        Merck, Inc
## GSM1523728        Merck, Inc
## GSM1523729        Merck, Inc
## GSM1523730        Merck, Inc
## GSM1523731        Merck, Inc
## GSM1523732        Merck, Inc
##                                                              contact_address
## GSM1523727 Mail Stop WP53B-120, 770 Sumneytown Pike, Building 53, P.O. Box 4
## GSM1523728 Mail Stop WP53B-120, 770 Sumneytown Pike, Building 53, P.O. Box 4
## GSM1523729 Mail Stop WP53B-120, 770 Sumneytown Pike, Building 53, P.O. Box 4
## GSM1523730 Mail Stop WP53B-120, 770 Sumneytown Pike, Building 53, P.O. Box 4
## GSM1523731 Mail Stop WP53B-120, 770 Sumneytown Pike, Building 53, P.O. Box 4
## GSM1523732 Mail Stop WP53B-120, 770 Sumneytown Pike, Building 53, P.O. Box 4
##            contact_city contact_state contact_zip/postal_code contact_country
## GSM1523727   West Point            PA                   19486             USA
## GSM1523728   West Point            PA                   19486             USA
## GSM1523729   West Point            PA                   19486             USA
## GSM1523730   West Point            PA                   19486             USA
## GSM1523731   West Point            PA                   19486             USA
## GSM1523732   West Point            PA                   19486             USA
##                                                                                                                  supplementary_file
## GSM1523727 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1523nnn/GSM1523727/suppl/GSM1523727_INT201201-10-1_107T_HG-U133_Plus_2_.CEL.gz
## GSM1523728 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1523nnn/GSM1523728/suppl/GSM1523728_INT201201-10-1_108T_HG-U133_Plus_2_.CEL.gz
## GSM1523729 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1523nnn/GSM1523729/suppl/GSM1523729_INT201201-10-1_109T_HG-U133_Plus_2_.CEL.gz
## GSM1523730 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1523nnn/GSM1523730/suppl/GSM1523730_INT201201-10-1_132N_HG-U133_Plus_2_.CEL.gz
## GSM1523731 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1523nnn/GSM1523731/suppl/GSM1523731_INT201201-10-1_133N_HG-U133_Plus_2_.CEL.gz
## GSM1523732 ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1523nnn/GSM1523732/suppl/GSM1523732_INT201201-10-1_134N_HG-U133_Plus_2_.CEL.gz
##            data_row_count patient:ch1            tissue:ch1
## GSM1523727          54675         107         Gastric tumor
## GSM1523728          54675         108         Gastric tumor
## GSM1523729          54675         109         Gastric tumor
## GSM1523730          54675         132 normal gastric tissue
## GSM1523731          54675         133 normal gastric tissue
## GSM1523732          54675         134 normal gastric tissue</code></pre>
<pre class="r"><code>phenotype&lt;-my.pdata %&gt;% 
  select(2,1)
row.names(phenotype)&lt;-NULL

#extract Normal or Tumor tissues
phenotype&lt;-phenotype %&gt;% 
  mutate(Tissue.type=substr(title,1,1)) %&gt;% 
  select(1,3)
colnames(phenotype)[1]&lt;-&quot;Patient.ID&quot;

# merge expression and phenotype dataframe
acrg.merged&lt;-phenotype %&gt;% 
  left_join(GOI)
create_dt(acrg.merged)</code></pre>
<div id="htmlwidget-a7f7dced60ed1065ee62" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a7f7dced60ed1065ee62">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400"],["GSM1523727","GSM1523728","GSM1523729","GSM1523730","GSM1523731","GSM1523732","GSM1523733","GSM1523734","GSM1523735","GSM1523736","GSM1523737","GSM1523738","GSM1523739","GSM1523740","GSM1523741","GSM1523742","GSM1523743","GSM1523744","GSM1523745","GSM1523746","GSM1523747","GSM1523748","GSM1523749","GSM1523750","GSM1523751","GSM1523752","GSM1523753","GSM1523754","GSM1523755","GSM1523756","GSM1523757","GSM1523758","GSM1523759","GSM1523760","GSM1523761","GSM1523762","GSM1523763","GSM1523764","GSM1523765","GSM1523766","GSM1523767","GSM1523768","GSM1523769","GSM1523770","GSM1523771","GSM1523772","GSM1523773","GSM1523774","GSM1523775","GSM1523776","GSM1523777","GSM1523778","GSM1523779","GSM1523780","GSM1523781","GSM1523782","GSM1523783","GSM1523784","GSM1523785","GSM1523786","GSM1523787","GSM1523788","GSM1523789","GSM1523790","GSM1523791","GSM1523792","GSM1523793","GSM1523794","GSM1523795","GSM1523796","GSM1523797","GSM1523798","GSM1523799","GSM1523800","GSM1523801","GSM1523802","GSM1523803","GSM1523804","GSM1523805","GSM1523806","GSM1523807","GSM1523808","GSM1523809","GSM1523810","GSM1523811","GSM1523812","GSM1523813","GSM1523814","GSM1523815","GSM1523816","GSM1523817","GSM1523818","GSM1523819","GSM1523820","GSM1523821","GSM1523822","GSM1523823","GSM1523824","GSM1523825","GSM1523826","GSM1523827","GSM1523828","GSM1523829","GSM1523830","GSM1523831","GSM1523832","GSM1523833","GSM1523834","GSM1523835","GSM1523836","GSM1523837","GSM1523838","GSM1523839","GSM1523840","GSM1523841","GSM1523842","GSM1523843","GSM1523844","GSM1523845","GSM1523846","GSM1523847","GSM1523848","GSM1523849","GSM1523850","GSM1523851","GSM1523852","GSM1523853","GSM1523854","GSM1523855","GSM1523856","GSM1523857","GSM1523858","GSM1523859","GSM1523860","GSM1523861","GSM1523862","GSM1523863","GSM1523864","GSM1523865","GSM1523866","GSM1523867","GSM1523868","GSM1523869","GSM1523870","GSM1523871","GSM1523872","GSM1523873","GSM1523874","GSM1523875","GSM1523876","GSM1523877","GSM1523878","GSM1523879","GSM1523880","GSM1523881","GSM1523882","GSM1523883","GSM1523884","GSM1523885","GSM1523886","GSM1523887","GSM1523888","GSM1523889","GSM1523890","GSM1523891","GSM1523892","GSM1523893","GSM1523894","GSM1523895","GSM1523896","GSM1523897","GSM1523898","GSM1523899","GSM1523900","GSM1523901","GSM1523902","GSM1523903","GSM1523904","GSM1523905","GSM1523906","GSM1523907","GSM1523908","GSM1523909","GSM1523910","GSM1523911","GSM1523912","GSM1523913","GSM1523914","GSM1523915","GSM1523916","GSM1523917","GSM1523918","GSM1523919","GSM1523920","GSM1523921","GSM1523922","GSM1523923","GSM1523924","GSM1523925","GSM1523926","GSM1523927","GSM1523928","GSM1523929","GSM1523930","GSM1523931","GSM1523932","GSM1523933","GSM1523934","GSM1523935","GSM1523936","GSM1523937","GSM1523938","GSM1523939","GSM1523940","GSM1523941","GSM1523942","GSM1523943","GSM1523944","GSM1523945","GSM1523946","GSM1523947","GSM1523948","GSM1523949","GSM1523950","GSM1523951","GSM1523952","GSM1523953","GSM1523954","GSM1523955","GSM1523956","GSM1523957","GSM1523958","GSM1523959","GSM1523960","GSM1523961","GSM1523962","GSM1523963","GSM1523964","GSM1523965","GSM1523966","GSM1523967","GSM1523968","GSM1523969","GSM1523970","GSM1523971","GSM1523972","GSM1523973","GSM1523974","GSM1523975","GSM1523976","GSM1523977","GSM1523978","GSM1523979","GSM1523980","GSM1523981","GSM1523982","GSM1523983","GSM1523984","GSM1523985","GSM1523986","GSM1523987","GSM1523988","GSM1523989","GSM1523990","GSM1523991","GSM1523992","GSM1523993","GSM1523994","GSM1523995","GSM1523996","GSM1523997","GSM1523998","GSM1523999","GSM1524000","GSM1524001","GSM1524002","GSM1524003","GSM1524004","GSM1524005","GSM1524006","GSM1524007","GSM1524008","GSM1524009","GSM1524010","GSM1524011","GSM1524012","GSM1524013","GSM1524014","GSM1524015","GSM1524016","GSM1524017","GSM1524018","GSM1524019","GSM1524020","GSM1524021","GSM1524022","GSM1524023","GSM1524024","GSM1524025","GSM1524026","GSM1524027","GSM1524028","GSM1524029","GSM1524030","GSM1524031","GSM1524032","GSM1524033","GSM1524034","GSM1524035","GSM1524036","GSM1524037","GSM1524038","GSM1524039","GSM1524040","GSM1524041","GSM1524042","GSM1524043","GSM1524044","GSM1524045","GSM1524046","GSM1524047","GSM1524048","GSM1524049","GSM1524050","GSM1524051","GSM1524052","GSM1524053","GSM1524054","GSM1524055","GSM1524056","GSM1524057","GSM1524058","GSM1524059","GSM1524060","GSM1524061","GSM1524062","GSM1524063","GSM1524064","GSM1524065","GSM1524066","GSM1524067","GSM1524068","GSM1524069","GSM1524070","GSM1524071","GSM1524072","GSM1524073","GSM1524074","GSM1524075","GSM1524076","GSM1524077","GSM1524078","GSM1524079","GSM1524080","GSM1524081","GSM1524082","GSM1524083","GSM1524084","GSM1524085","GSM1524086","GSM1524087","GSM1524088","GSM1524089","GSM1524090","GSM1524091","GSM1524092","GSM1524093","GSM1524094","GSM1524095","GSM1524096","GSM1524097","GSM1524098","GSM1524099","GSM1524100","GSM1524101","GSM1524102","GSM1524103","GSM1524104","GSM1524105","GSM1524106","GSM1524107","GSM1524108","GSM1524109","GSM1524110","GSM1524111","GSM1524112","GSM1524113","GSM1524114","GSM1524115","GSM1524116","GSM1524117","GSM1524118","GSM1524119","GSM1524120","GSM1524121","GSM1524122","GSM1524123","GSM1524124","GSM1524125","GSM1524126"],["T","T","T","N","N","N","N","N","N","N","N","N","N","N","N","N","N","T","T","T","T","T","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","T","N","N","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","N","T","N","T","T","T","T","N","T","T","T","T","N","N","N","N","N","N","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","T","N","N","N","N","N","T","T","T","T","T","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N","N"],[2.957374811,3.12856555,3.052886724,2.916911125,2.789613485,3.088736296,2.728603363,2.688107729,2.815227747,2.719876051,2.635167599,2.737114668,2.815265417,2.817178249,2.749531269,2.822958231,2.792710304,3.147130251,2.963590622,3.235590696,3.085235357,2.995404005,2.845135212,2.636401892,2.715661049,2.814172029,2.743992329,2.8425107,2.809187412,2.682807922,2.792397022,2.822592497,2.817968607,2.67236805,2.810774565,2.680358887,2.746968269,2.796033859,3.076695204,2.586037159,2.731681585,2.999129295,2.950501919,3.210108519,2.843728542,2.874548197,2.913945436,2.718482733,3.119673252,3.150796652,3.159020662,3.006353378,2.926577568,3.155986547,2.938718319,3.306967974,2.929345846,2.911279202,2.982079029,2.961564779,2.996509075,3.032979727,2.895038366,2.97686553,2.996754169,3.306386948,3.062892914,2.999716759,2.795334578,3.035706997,2.964684725,2.978999615,3.18087554,3.20947361,3.177567244,3.234570503,3.047708988,2.987613201,2.99837327,3.055806637,3.008315086,3.04057169,3.2911129,3.210545063,3.057314873,3.262635231,3.271958351,3.109438181,3.147539616,3.217730522,3.424568415,2.917867661,3.017991304,3.00960803,3.184214115,2.923691034,3.077348232,3.091773748,3.185035706,2.945376635,2.859711647,2.997797251,3.095702171,3.072796822,3.204699039,2.80539012,3.039045334,2.968037844,3.038310766,2.97823,3.063546181,3.02499032,2.851779699,3.128544569,3.385202885,3.047880411,3.042798996,3.281862259,2.938891888,2.998197317,3.080500126,3.068455935,3.030682564,2.871552944,3.122219801,2.853273392,2.94675827,3.009040594,2.970867395,3.202682257,2.881159544,2.945680618,2.891820908,3.067465782,2.964907408,2.855398417,3.159384966,3.070234776,3.089245081,3.128676891,2.955404043,3.202615976,2.974518299,3.443379641,3.131870747,3.015122414,3.383071423,3.190321684,3.20034337,3.120811224,3.31304574,3.102568626,3.003628969,3.415651798,3.101578236,3.181128502,3.18642664,3.097692013,3.043337822,3.004098415,3.08466959,3.342209578,2.897662878,2.97573781,3.063239098,2.921816111,3.026627779,2.912772179,3.246608496,3.066005468,2.854987383,3.315679789,2.893266678,2.911756992,3.074241638,2.950635195,3.137265444,3.012909889,3.036272764,3.095750332,2.848033667,3.186691523,3.008666515,3.108258009,2.874035835,2.770051479,2.702991724,2.910713196,2.681807756,2.678904772,2.827189445,3.084076405,2.937488079,2.626896143,3.195312977,2.95795393,2.947173595,2.838231802,3.093760729,3.014595509,3.013391495,3.22127986,3.078682184,3.022561073,3.378886938,3.182952642,3.083880663,3.185391188,3.004637241,2.951573372,2.936838627,2.978689671,2.851360559,3.12207222,3.157214642,3.201673746,3.207176447,3.385419607,2.928642511,2.77918601,3.232448101,2.961089373,2.907426357,3.119022846,3.113481045,2.89683485,3.249185324,2.618034601,2.608339071,3.11122632,2.973438978,2.912148952,3.28081131,3.19568634,3.212038279,2.939175844,2.841187954,3.088805437,2.859143972,3.223212481,2.83851409,2.941824198,2.938583374,2.784324408,2.797530413,2.8247087,3.126699209,2.942360878,3.148289204,3.026016712,3.258667707,3.061161757,3.287966967,2.879472256,3.052657843,2.982475758,3.244212151,3.229645491,3.214660168,2.813073397,2.844922543,2.860682487,2.889500618,3.15059185,3.07360363,3.009462595,3.106150866,2.844342709,3.059858561,2.752822876,3.182507038,3.213233471,2.665421247,2.989789724,2.95788002,3.019896984,2.836535454,3.209268808,3.217390537,2.849848986,3.183789492,3.169156551,2.968491077,2.817076206,2.974882364,3.093784809,2.939147711,3.30209446,3.09094882,2.918070793,3.104841471,3.215963602,2.971065521,2.999756098,2.934294224,2.972426891,3.287458181,2.942452192,2.97755456,3.102526426,2.806434631,3.141832113,2.667832136,2.838275433,3.123773336,3.175128937,3.117863894,2.731698275,2.889741182,3.038846493,3.049394846,3.023415804,3.000003576,2.954527378,3.085741282,3.154393911,3.061396599,3.107219696,2.789932728,3.200454473,2.938494921,2.926928759,3.072450638,3.174608231,2.979375839,2.872581005,2.997127056,3.233898878,3.24414897,3.042684793,3.052739143,2.908396959,3.018183947,3.222173929,3.118667841,2.951304913,2.831955194,2.884925842,3.012903929,2.834344387,2.848714352,3.097692013,3.118122816,3.039960384,3.087878466,3.189382792,2.759836197,2.990402699,2.952914238,2.970369577,2.935771465,2.74992466,2.88511157,2.824718952,2.767581701,2.777864695,2.763994694,2.994106054,2.830463886,2.678862572,2.992287159,2.892502308,2.772145271,2.852181911,2.907083273,2.849184036,2.643030882,2.978168249,2.970688343,2.942844152,2.8031919,2.796320677,2.867218256,2.90090847,2.758628368,2.699540138,2.895653009,2.703393936,2.905613184,2.933779478,2.83579731,2.84437418,2.742387772,2.771556616,2.788948774,2.691838503,2.834642172,2.914143801,2.852364063,2.788648844,2.778459549,2.852918863,3.000881195,2.96034956,2.895446301,2.891337395,2.915441275,2.632597446,2.892452002,2.886523247],[1.999631166,1.689933658,1.806846499,1.659222841,1.710084915,1.715045214,1.715170741,1.611777782,1.788779497,1.732818365,1.730774045,1.753004909,1.627176166,1.649507761,1.425686598,1.576099515,1.682697177,1.838928699,1.969595551,1.801892042,1.665202498,1.808223963,1.702522039,1.783264041,1.750326276,1.536372185,1.718086243,1.700418472,1.726334095,1.775270581,1.732086658,1.798879385,1.711128831,1.693511367,1.663894773,1.71475327,1.728121638,1.687718987,1.852277517,1.738290787,1.627220035,1.721357584,1.557868481,1.759343982,1.578724742,1.503576756,1.707289219,1.664178491,1.796833992,1.858202696,1.835614324,1.680958152,1.756504059,1.668644786,2.004040003,1.853693008,1.664421916,1.750106812,1.879280567,1.746286869,2.096833706,1.607367754,1.949021339,1.635404706,1.77926302,1.841823697,1.662317991,1.611137867,1.905258298,1.662628412,1.724052668,1.641749859,1.678266287,1.857214808,1.550714135,1.870457649,1.917329907,1.740756273,1.777047515,1.754122972,1.743704796,1.705020428,1.857317805,1.640807033,1.873301268,1.622694612,1.699964881,1.942359209,2.077547789,1.915689588,1.769298077,1.657966733,1.689486265,1.841080904,1.644754767,1.772178411,1.790851474,1.817982793,1.583831906,1.775664926,1.690455317,1.79350245,1.937746644,1.862377763,1.923757434,1.725263119,1.523513079,1.563017368,1.555177212,1.755025983,1.815087795,1.654911518,1.74778688,1.637545943,1.888604999,1.670453668,1.665693879,1.807665944,1.751270294,1.744133592,2.148107052,1.593389153,1.809669733,1.761662245,1.710757375,1.737655282,1.726963878,1.770779729,2.217208624,1.918112874,1.781154037,1.747822523,1.917981863,1.957172394,1.657834768,1.644781828,1.661821365,1.540039659,2.124090195,1.87273562,1.85440588,1.810659051,1.790277481,1.749420166,1.837577105,1.734749198,1.581658363,1.892358899,2.261018038,1.888759732,2.183561087,1.805012584,2.172904968,1.817941546,1.697178602,1.900523186,1.977327585,1.796301365,2.056133986,1.63303709,1.924376965,2.130399704,1.927663684,1.795584202,1.746777177,1.703609347,2.094990015,1.943907261,1.713058233,1.802708268,1.570431948,2.269286394,1.784659624,1.768383622,1.752551436,1.633904934,1.713090062,1.819737792,1.815981746,1.688702345,1.694081545,1.575130701,1.601122499,1.905029535,1.791112185,1.830586314,1.812950492,1.67119205,1.843350291,1.68662703,1.805656195,1.779310584,1.823068976,1.776023984,1.854514003,1.966518164,1.831668496,1.928054452,1.741906166,1.844726801,1.805217266,1.857804775,1.772027493,1.895392656,1.721930146,1.964113474,1.677382827,1.8531394,1.459932923,1.856469154,1.656533718,1.773256302,1.77469492,1.666191339,1.752809763,1.945147157,1.92158854,1.79148221,1.84341681,1.854770184,1.97617197,1.802625775,1.800246358,1.932660341,1.717374682,1.740918994,2.044832706,1.776778102,1.895971537,1.773994446,1.898185611,1.721430182,2.075010061,2.165514708,1.843223214,1.612985253,1.75635922,1.86720717,1.722007155,1.878753185,1.667879581,1.803445816,1.858840346,1.74158442,1.594932199,1.738686919,2.04752183,2.076325178,1.666353941,1.618975163,1.801205516,2.125076294,1.956751823,1.653876543,1.624467731,1.632444382,2.167976141,1.480919123,1.741520762,1.823555231,1.755066276,1.643834472,1.788356066,1.686588287,1.780757308,1.952799678,1.762620687,1.771220088,1.652367473,1.713201761,1.790102005,1.849505782,1.695103765,1.888008714,1.624637842,1.874225736,2.090979576,1.842685223,1.684231162,1.745157242,1.88411808,1.54759562,1.803656816,1.725294471,1.949798584,1.818771243,1.734773278,1.764786243,1.778705239,1.662485719,1.947886229,1.828315616,1.776421905,1.788346767,1.77256155,1.818954229,1.924840212,1.9106704,1.735651374,1.789448142,1.902002931,1.904873967,1.786220908,1.70116961,1.987693548,1.871449947,1.783687949,1.802808762,1.793010831,1.984903932,1.71742487,1.780708671,1.605951071,1.811505198,1.939885736,1.755030394,1.727571726,2.07227993,1.722268701,1.816555262,1.910726905,1.759799361,2.249561071,1.902220488,1.712842941,1.905756116,2.053143263,1.916097522,1.887490988,2.034775972,1.772382379,1.738080025,1.762629986,2.066730499,1.879084706,1.725506902,1.893591642,1.785773396,1.829931498,1.911271334,1.80541718,2.167268991,1.884902835,1.855713248,1.905139327,1.801498055,1.709479213,1.720911384,1.775472283,1.67495203,1.673384547,1.689043403,1.592502475,1.735837936,1.793655872,1.644643903,1.775362253,1.763368368,1.685159326,1.865243554,1.700535893,1.833056211,1.714719534,1.835996151,1.662488461,1.678172708,1.838349819,1.662653565,1.69565165,1.671333313,1.780729651,1.836345196,1.733565807,1.658667445,1.677853942,1.668392181,1.686002254,1.719659686,1.737759829,1.635022402,1.78942883,1.770885348,1.620204329,1.894323111,1.67212081,1.851933241,1.613937616,1.740922689,1.792586327,1.831160665,1.758775234,1.704298496,1.68480432,1.664698005,1.641216755,1.776688099,1.797309756,1.821109533,1.688664198,1.542821765],[1.758548856,1.830103755,1.79157424,1.750806332,1.744395971,1.766944528,1.715903759,1.686532021,1.621681333,1.652005672,1.707924962,1.709467173,1.626820564,1.736498475,1.711333275,1.679805756,1.634628654,1.851753473,1.717695475,1.761043787,1.734171152,1.666710973,1.758925796,1.751577973,1.671687007,1.726404309,1.709783912,1.622655869,1.686190963,1.674128056,1.698174834,1.763122439,1.771084428,1.656049371,1.674028039,1.698989511,1.618973017,1.684686065,1.855855703,1.694189548,1.654814601,1.686451674,1.685370088,1.821693897,1.690219879,1.763311505,1.701994658,1.683205605,1.689013958,1.772204518,1.793810964,1.662600994,1.707484365,1.772475839,1.823301435,1.692118526,1.647201896,1.653480172,1.682448268,1.75089097,1.897150874,1.705628514,1.65963161,1.7332412,1.737791061,1.749219656,1.584639668,1.729683995,1.710935235,1.785650253,1.780675292,1.685570836,1.792230248,1.885370374,1.737324476,1.691612124,1.700615406,1.675217748,1.733422279,1.814955592,1.755568147,1.597625852,1.891177297,1.749339581,1.792098045,1.643388987,1.704729915,1.69510591,2.021564245,1.893461585,1.686021447,1.815715313,1.767421126,1.879446864,1.932706475,1.742809534,1.794052601,1.725013018,1.994132757,1.715401053,1.62768209,1.705903292,1.721472502,1.675522447,1.811334848,1.703554034,1.684233665,1.724266171,1.758895159,1.695086479,1.720859051,1.92911303,1.712578654,1.615649104,1.559627771,1.86244452,1.79372251,1.692676663,1.724153161,1.720668197,1.778208017,1.666377664,1.631718636,1.752571702,1.844559431,1.645853877,1.627763033,1.71106267,1.973517418,1.82411027,1.671358943,1.880775452,1.756843686,1.726024151,1.691114902,1.711549401,1.824608088,1.699100614,2.006079435,2.033657551,1.720910549,1.772354364,1.641874194,1.680268168,1.750204682,1.676889181,1.784520268,1.874706507,1.914802194,1.705708861,1.832034469,1.640307426,1.748783112,1.888203502,1.829136491,1.598534822,1.796661735,1.627461195,1.96785295,1.656455159,1.776043296,1.989938617,1.767684817,1.762513876,1.673834324,1.667478681,1.906075239,1.921959519,1.875171542,1.737155676,1.843141437,1.775982738,1.753883839,1.806489468,1.98863101,1.755125523,1.672229171,1.608310461,2.038288355,1.655387759,1.750592351,1.73109889,1.588617563,1.714745641,1.759968281,1.716706276,1.787609816,1.873993397,1.779178262,1.77533555,1.717286706,1.841808796,1.693056107,1.732866526,1.842051864,1.812234282,1.816965938,1.705434799,1.910307169,1.848892331,1.84317565,1.811853766,1.85754931,2.018503666,1.741691828,1.679985762,1.737810969,2.075521946,1.699348569,1.67100358,1.761135936,1.811118126,1.749459982,1.823159337,1.788239717,1.989214659,1.955542922,1.707691073,1.813322186,1.863623142,1.891790986,1.707470655,1.724795699,1.935398579,1.645061851,1.661957979,1.846655607,1.814749718,1.736379504,1.693524241,1.771361589,1.616713166,1.875813603,1.786052704,1.931678653,1.734116673,1.710356474,1.773177147,1.560267091,1.921811581,1.749869585,1.826864004,1.763747692,1.8307091,1.818864822,1.990679502,1.754351616,1.900218368,1.755191445,1.7557863,1.839985847,1.768248916,1.784755468,1.638425231,1.756263137,1.67266798,2.010956287,1.796910644,1.670746565,1.687574387,1.757026911,1.888431311,1.724116087,1.780285358,1.706837654,1.770494699,1.683920503,1.779313326,1.639549613,1.738237858,2.031646013,1.838929892,1.873702288,1.780515313,1.972054124,1.77727294,1.803524613,1.710367799,1.773458123,1.712549329,1.792763352,1.822919011,1.737038255,1.722760081,1.786082864,1.833286166,1.832191706,2.108191252,1.930264831,1.724506021,1.852566242,1.760377407,1.786398411,1.617211699,1.746062756,1.77879715,1.74303329,1.617865801,1.725845933,1.781158447,1.740361214,1.689473391,1.75270474,1.728888035,1.814137936,1.841030478,1.810917616,1.78456533,1.771012068,1.754203916,1.575269818,1.683729649,1.638297677,1.587837458,1.727417946,1.589079261,1.738102913,1.787741303,1.640357971,1.708609819,1.677488089,1.635865569,1.741361856,1.760099411,1.769269228,1.814956427,1.863052964,1.907531261,1.731840253,2.029956102,1.737115622,1.789052248,1.662871718,1.943005681,1.925180078,1.81296277,1.883307576,1.875577688,1.902134061,1.836507559,1.782127738,2.044288874,1.793416858,1.719825387,2.023029089,1.917829871,1.659417868,1.803114653,1.752658129,1.850113273,1.840152383,1.645513415,1.689228296,1.708601952,1.679570317,1.651222348,1.755802393,1.898722291,1.769207239,1.768238187,1.723926187,1.85790658,1.679285049,1.656700253,1.730875492,1.65122664,1.777728796,1.805466294,1.720599174,1.8060987,1.752981663,1.787553787,1.662734747,1.696356058,1.602196097,1.765600324,1.687205911,1.652005672,1.62503469,1.822181463,1.749342799,1.668096542,1.806867838,1.728935719,1.827988625,1.770244122,1.710799336,1.702970624,1.757380605,1.726594567,1.686441422,1.763129354,1.838609099,1.882244229,1.730498075,1.697542191,1.766449213,1.662563562,1.664028168,1.754217386]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Patient.ID<\/th>\n      <th>Tissue.type<\/th>\n      <th>PRMT1<\/th>\n      <th>CAMKMT<\/th>\n      <th>FAM86A<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"lengthMenu":[[10,25,50,-1],["10","25","50","All"]],"columnDefs":[{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="plotting" class="section level3" number="1.1.2">
<h3><span class="header-section-number">1.1.2</span> Plotting</h3>
<pre class="r"><code>acrg&lt;-acrg.merged %&gt;% 
  mutate(Tissue2 = case_when(Tissue.type == &#39;N&#39; ~ &#39;Normal&#39;,
                             Tissue.type == &#39;T&#39; ~ &#39;Tumor&#39;))
acrg$Tissue2 &lt;- factor(acrg$Tissue2, levels = c(&quot;Normal&quot;, &quot;Tumor&quot;))

GSEacrg1&lt;-ggplot(acrg,aes(x=Tissue2,y=FAM86A,fill=Tissue2))+
  labs(y=expression(&#39;FAM86A expression (log&#39;[10]*&quot;)&quot;),x=NULL)+
  stat_boxplot(geom =&#39;errorbar&#39;,width=.2) +
  geom_jitter(colour=&#39;gray90&#39;,alpha=.8,width = .2)+
  geom_boxplot(width=.6)+
  theme_Publication()+
  scale_fill_aziz()+
  scale_y_continuous(expand=expand_scale(mult = c(0.1,0.2)))+
  theme(axis.text.x = element_text(),
        legend.position = &#39;none&#39;,
        axis.title.x = element_blank(),
        panel.grid.major.x = element_blank(),
        aspect.ratio = 1.85)
GSEacrg2&lt;-ggplot(acrg,aes(x=Tissue2,y=CAMKMT,fill=Tissue2))+
  labs(y=expression(&#39;CAMKMT expression (log&#39;[10]*&quot;)&quot;),x=NULL)+
  stat_boxplot(geom =&#39;errorbar&#39;,width=.2) +
  geom_jitter(colour=&#39;gray90&#39;,alpha=.8,width = .2)+
  geom_boxplot(width=.6)+
  theme_Publication()+
  scale_fill_aziz()+
  scale_y_continuous(expand=expand_scale(mult = c(0.1,0.2)))+
  theme(axis.text.x = element_text(),
        legend.position = &#39;none&#39;,
        axis.title.x = element_blank(),
        panel.grid.major.x = element_blank(),
        aspect.ratio = 1.85)
GSEacrg3&lt;-ggplot(acrg,aes(x=Tissue2,y=PRMT1,fill=Tissue2))+
  labs(y=expression(&#39;PRMT1 expression (log&#39;[10]*&quot;)&quot;),x=NULL)+
  stat_boxplot(geom =&#39;errorbar&#39;,width=.2) +
  geom_jitter(colour=&#39;gray90&#39;,alpha=.8,width = .2)+
  geom_boxplot(width=.6)+
  theme_Publication()+
  scale_fill_aziz()+
  scale_y_continuous(expand=expand_scale(mult = c(0.1,0.2)))+
  theme(axis.text.x = element_text(),
        legend.position = &#39;none&#39;,
        axis.title.x = element_blank(),
        panel.grid.major.x = element_blank(),
        aspect.ratio = 1.85)
library(gridExtra)
grid.arrange(GSEacrg1,GSEacrg2,GSEacrg3,ncol=3) </code></pre>
<p><img src="Gene-Expression-Note-1_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
<div id="statistical-analysis" class="section level3" number="1.1.3">
<h3><span class="header-section-number">1.1.3</span> Statistical
analysis</h3>
<pre class="r"><code>wilcox.test(FAM86A ~ Tissue2, data = acrg,
            exact = FALSE)</code></pre>
<pre><code>## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  FAM86A by Tissue2
## W = 11848, p-value = 0.001646
## alternative hypothesis: true location shift is not equal to 0</code></pre>
<p><br></p>
</div>
</div>
<div id="unify-gtex-nat-and-tumor-tcga" class="section level2"
number="1.2">
<h2><span class="header-section-number">1.2</span> Unify GTEx NAT and
Tumor TCGA</h2>
<ul>
<li>datasets retrieve from [Nature scientific data] (<a
href="https://doi.org/10.1038/sdata.2018.61"
class="uri">https://doi.org/10.1038/sdata.2018.61</a>)</li>
<li>first load the all 3 datasets (tcga normal, tcga tumor, and gtex
-&gt; FPKM normalized Combat batch correction)</li>
</ul>
<pre class="r"><code>stomach.rsem.fpkm.gtex &lt;- read.delim(&quot;docs/Gene_Exp_1/stomach-rsem-fpkm-gtex.txt&quot;)
stad.rsem.fpkm.tcga.t &lt;- read.delim(&quot;docs/Gene_Exp_1/stad-rsem-fpkm-tcga-t.txt&quot;)
stad.rsem.fpkm.tcga &lt;- read.delim(&quot;docs/Gene_Exp_1/stad-rsem-fpkm-tcga.txt&quot;)

df.gtex&lt;-stomach.rsem.fpkm.gtex
df.t.tcga&lt;-stad.rsem.fpkm.tcga.t
df.n.tcga&lt;-stad.rsem.fpkm.tcga


library(tidyverse)
library(dplyr)
Sys.setenv(LANG=&#39;en&#39;)

############GTEx###############
colnames(df.gtex)[1:5]</code></pre>
<pre><code>## [1] &quot;Hugo_Symbol&quot;              &quot;Entrez_Gene_Id&quot;          
## [3] &quot;GTEX.QCQG.0526.SM.48U2A&quot;  &quot;GTEX.QDVJ.1426.SM.48U1Y&quot; 
## [5] &quot;GTEX.13111.1226.SM.5GCNC&quot;</code></pre>
<pre class="r"><code>df.gtex.FAM86A&lt;-df.gtex %&gt;% 
  filter(Hugo_Symbol == &#39;FAM86A&#39;)

head(df.gtex.FAM86A)[2]</code></pre>
<pre><code>##   Entrez_Gene_Id
## 1         196483</code></pre>
<pre class="r"><code>dim(df.gtex.FAM86A)</code></pre>
<pre><code>## [1]   1 194</code></pre>
<pre class="r"><code>#transposing
df.gtex.FAM86A&lt;-df.gtex.FAM86A %&gt;% 
  gather(Patients,FPKM,3:ncol(.))

dim(df.gtex.FAM86A)</code></pre>
<pre><code>## [1] 192   4</code></pre>
<pre class="r"><code>head(df.gtex.FAM86A)[1:4]</code></pre>
<pre><code>##   Hugo_Symbol Entrez_Gene_Id                 Patients   FPKM
## 1      FAM86A         196483  GTEX.QCQG.0526.SM.48U2A 224.97
## 2      FAM86A         196483  GTEX.QDVJ.1426.SM.48U1Y 220.32
## 3      FAM86A         196483 GTEX.13111.1226.SM.5GCNC 121.79
## 4      FAM86A         196483  GTEX.S4Z8.1226.SM.4AD6W 185.11
## 5      FAM86A         196483  GTEX.ZDYS.1926.SM.5HL59 171.45
## 6      FAM86A         196483 GTEX.144GM.1726.SM.5O9AS 185.11</code></pre>
<pre class="r"><code>df.gtex.FAM86A.log2&lt;-df.gtex.FAM86A %&gt;% 
  mutate(log2FPKM=log(FPKM,base = 2))
head(df.gtex.FAM86A.log2)[1:5]</code></pre>
<pre><code>##   Hugo_Symbol Entrez_Gene_Id                 Patients   FPKM log2FPKM
## 1      FAM86A         196483  GTEX.QCQG.0526.SM.48U2A 224.97 7.813589
## 2      FAM86A         196483  GTEX.QDVJ.1426.SM.48U1Y 220.32 7.783457
## 3      FAM86A         196483 GTEX.13111.1226.SM.5GCNC 121.79 6.928252
## 4      FAM86A         196483  GTEX.S4Z8.1226.SM.4AD6W 185.11 7.532239
## 5      FAM86A         196483  GTEX.ZDYS.1926.SM.5HL59 171.45 7.421644
## 6      FAM86A         196483 GTEX.144GM.1726.SM.5O9AS 185.11 7.532239</code></pre>
<pre class="r"><code>#######NAT###########
colnames(df.n.tcga)[1:5]</code></pre>
<pre><code>## [1] &quot;Hugo_Symbol&quot;                  &quot;Entrez_Gene_Id&quot;              
## [3] &quot;TCGA.BR.6564.11A.01R.1884.13&quot; &quot;TCGA.HU.A4GH.11A.11R.A36D.31&quot;
## [5] &quot;TCGA.IN.8663.11A.01R.2402.13&quot;</code></pre>
<pre class="r"><code>df.n.tcga.FAM86A&lt;-df.n.tcga %&gt;% 
  filter(Hugo_Symbol == &#39;FAM86A&#39;)

head(df.n.tcga.FAM86A)[2]</code></pre>
<pre><code>##   Entrez_Gene_Id
## 1         196483</code></pre>
<pre class="r"><code>dim(df.n.tcga.FAM86A)</code></pre>
<pre><code>## [1]  1 35</code></pre>
<pre class="r"><code>#transposing
df.n.tcga.FAM86A&lt;-df.n.tcga.FAM86A %&gt;% 
  gather(Patients,FPKM,3:ncol(.))

dim(df.n.tcga.FAM86A)</code></pre>
<pre><code>## [1] 33  4</code></pre>
<pre class="r"><code>head(df.n.tcga.FAM86A)[1:4]</code></pre>
<pre><code>##   Hugo_Symbol Entrez_Gene_Id                     Patients   FPKM
## 1      FAM86A         196483 TCGA.BR.6564.11A.01R.1884.13 220.32
## 2      FAM86A         196483 TCGA.HU.A4GH.11A.11R.A36D.31 202.66
## 3      FAM86A         196483 TCGA.IN.8663.11A.01R.2402.13 166.73
## 4      FAM86A         196483 TCGA.HU.8238.11A.01R.2343.13 163.28
## 5      FAM86A         196483 TCGA.BR.7716.11A.01R.2055.13 249.73
## 6      FAM86A         196483 TCGA.FP.7735.11A.01R.2055.13 256.78</code></pre>
<pre class="r"><code>df.n.tcga.FAM86A.log2&lt;-df.n.tcga.FAM86A %&gt;% 
  mutate(log2FPKM=log(FPKM,base = 2))
head(df.n.tcga.FAM86A.log2)[1:5]</code></pre>
<pre><code>##   Hugo_Symbol Entrez_Gene_Id                     Patients   FPKM log2FPKM
## 1      FAM86A         196483 TCGA.BR.6564.11A.01R.1884.13 220.32 7.783457
## 2      FAM86A         196483 TCGA.HU.A4GH.11A.11R.A36D.31 202.66 7.662918
## 3      FAM86A         196483 TCGA.IN.8663.11A.01R.2402.13 166.73 7.381370
## 4      FAM86A         196483 TCGA.HU.8238.11A.01R.2343.13 163.28 7.351204
## 5      FAM86A         196483 TCGA.BR.7716.11A.01R.2055.13 249.73 7.964225
## 6      FAM86A         196483 TCGA.FP.7735.11A.01R.2055.13 256.78 8.004389</code></pre>
<pre class="r"><code>#######Tumor###########
colnames(df.t.tcga)[1:5]</code></pre>
<pre><code>## [1] &quot;Hugo_Symbol&quot;                  &quot;Entrez_Gene_Id&quot;              
## [3] &quot;TCGA.BR.6457.01A.21R.1802.13&quot; &quot;TCGA.VQ.A8PD.01A.11R.A414.31&quot;
## [5] &quot;TCGA.BR.6803.01A.11R.1884.13&quot;</code></pre>
<pre class="r"><code>df.t.tcga.FAM86A&lt;-df.t.tcga %&gt;% 
  filter(Hugo_Symbol == &#39;FAM86A&#39;)

head(df.t.tcga.FAM86A)[2]</code></pre>
<pre><code>##   Entrez_Gene_Id
## 1         196483</code></pre>
<pre class="r"><code>dim(df.t.tcga.FAM86A)</code></pre>
<pre><code>## [1]   1 382</code></pre>
<pre class="r"><code>#transposing
df.t.tcga.FAM86A&lt;-df.t.tcga.FAM86A %&gt;% 
  gather(Patients,FPKM,3:ncol(.))

dim(df.t.tcga.FAM86A)</code></pre>
<pre><code>## [1] 380   4</code></pre>
<pre class="r"><code>head(df.t.tcga.FAM86A)[1:4]</code></pre>
<pre><code>##   Hugo_Symbol Entrez_Gene_Id                     Patients   FPKM
## 1      FAM86A         196483 TCGA.BR.6457.01A.21R.1802.13 325.29
## 2      FAM86A         196483 TCGA.VQ.A8PD.01A.11R.A414.31 215.77
## 3      FAM86A         196483 TCGA.BR.6803.01A.11R.1884.13 346.29
## 4      FAM86A         196483 TCGA.VQ.A8E3.01A.11R.A39E.31 277.20
## 5      FAM86A         196483 TCGA.CG.5718.01A.11R.1602.13 351.14
## 6      FAM86A         196483 TCGA.VQ.A8E2.01A.11R.A36D.31 356.05</code></pre>
<pre class="r"><code>df.t.tcga.FAM86A.log2&lt;-df.t.tcga.FAM86A %&gt;% 
  mutate(log2FPKM=log(FPKM,base = 2))
head(df.t.tcga.FAM86A.log2)[1:5]</code></pre>
<pre><code>##   Hugo_Symbol Entrez_Gene_Id                     Patients   FPKM log2FPKM
## 1      FAM86A         196483 TCGA.BR.6457.01A.21R.1802.13 325.29 8.345583
## 2      FAM86A         196483 TCGA.VQ.A8PD.01A.11R.A414.31 215.77 7.753350
## 3      FAM86A         196483 TCGA.BR.6803.01A.11R.1884.13 346.29 8.435837
## 4      FAM86A         196483 TCGA.VQ.A8E3.01A.11R.A39E.31 277.20 8.114783
## 5      FAM86A         196483 TCGA.CG.5718.01A.11R.1602.13 351.14 8.455903
## 6      FAM86A         196483 TCGA.VQ.A8E2.01A.11R.A36D.31 356.05 8.475936</code></pre>
<pre class="r"><code>###############combine the 3 data to 1###########
s.gtex&lt;-df.gtex.FAM86A.log2 %&gt;% 
  mutate(data=&#39;GTEx&#39;) %&gt;% 
  dplyr::select(&#39;data&#39;,&#39;log2FPKM&#39;)
head(s.gtex)[1:2]</code></pre>
<pre><code>##   data log2FPKM
## 1 GTEx 7.813589
## 2 GTEx 7.783457
## 3 GTEx 6.928252
## 4 GTEx 7.532239
## 5 GTEx 7.421644
## 6 GTEx 7.532239</code></pre>
<pre class="r"><code>dim(s.gtex)</code></pre>
<pre><code>## [1] 192   2</code></pre>
<pre class="r"><code>s.nor&lt;-df.n.tcga.FAM86A.log2 %&gt;% 
  mutate(data=&#39;NAT&#39;) %&gt;% 
  dplyr::select(&#39;data&#39;,&#39;log2FPKM&#39;)

s.tumor&lt;-df.t.tcga.FAM86A.log2 %&gt;% 
  mutate(data=&#39;Tumor&#39;) %&gt;% 
  dplyr::select(&#39;data&#39;,&#39;log2FPKM&#39;)

#binding rows
unify.FAM86A&lt;-rbind(s.gtex,s.nor,s.tumor)

head(unify.FAM86A)[1:2]</code></pre>
<pre><code>##   data log2FPKM
## 1 GTEx 7.813589
## 2 GTEx 7.783457
## 3 GTEx 6.928252
## 4 GTEx 7.532239
## 5 GTEx 7.421644
## 6 GTEx 7.532239</code></pre>
<pre class="r"><code>dim(unify.FAM86A)</code></pre>
<pre><code>## [1] 605   2</code></pre>
<pre class="r"><code>count(unify.FAM86A,data)</code></pre>
<pre><code>##    data   n
## 1  GTEx 192
## 2   NAT  33
## 3 Tumor 380</code></pre>
<div id="plotting-1" class="section level3" number="1.2.1">
<h3><span class="header-section-number">1.2.1</span> Plotting</h3>
<pre class="r"><code>a&lt;-unify.FAM86A %&gt;%
  mutate(&#39;source&#39;= case_when(data == &#39;GTEx&#39; ~ &#39;GTEx&#39;,
                             data == &#39;NAT&#39; ~ &#39;Normal TCGA&#39;,
                             data == &#39;Tumor&#39; ~ &#39;Tumor TCGA&#39;))

a %&gt;% 
  ggplot(aes(x=source,y=log2FPKM,fill=source))+
  
  
  
  
  stat_boxplot(geom =&#39;errorbar&#39;,width=.2) +
  geom_jitter(colour=&#39;gray90&#39;,alpha=.8,width = .2)+
  geom_boxplot(width=.6)+
  theme_Publication()+
  scale_fill_aziz()+
  scale_y_continuous(expand=expand_scale(mult = c(0.1,0.2)))+
  theme(axis.text.x = element_text(angle=45,hjust=1,vjust=1),
        legend.position = &#39;none&#39;,
        axis.title.x = element_blank(),
        panel.grid.major.x = element_blank(),
        aspect.ratio = 1.85)</code></pre>
<p><img src="Gene-Expression-Note-1_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
</div>
<div id="statistical-analysis-1" class="section level3" number="1.2.2">
<h3><span class="header-section-number">1.2.2</span> Statistical
analysis</h3>
<pre class="r"><code>kruskal.test(log2FPKM ~ source, data = a)</code></pre>
<pre><code>## 
##  Kruskal-Wallis rank sum test
## 
## data:  log2FPKM by source
## Kruskal-Wallis chi-squared = 55.539, df = 2, p-value = 8.706e-13</code></pre>
<pre class="r"><code>pairwise.wilcox.test(a$log2FPKM, a$source,paired=FALSE,
                     p.adjust.method = &quot;BH&quot;)</code></pre>
<pre><code>## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  a$log2FPKM and a$source 
## 
##             GTEx    Normal TCGA
## Normal TCGA 0.63    -          
## Tumor TCGA  2.1e-11 8.9e-05    
## 
## P value adjustment method: BH</code></pre>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
